core/runtime/src/main/java/io/quarkus/runtime/configuration/LocaleConverter.java

```diff
@@ -29,11 +29,6 @@ public Locale convert(final String value) {
             return null;
         }
 
-        Locale locale = Locale.forLanguageTag(NORMALIZE_LOCALE_PATTERN.matcher(localeValue).replaceAll("-"));
-        if (locale.getLanguage() == null || locale.getLanguage().isEmpty()) {
```

Comment:
I think I would prefer if we kept these tests but also check that the locale is not `Locale.ROOT`.

Reply:
Done